Pcap++/src/PcapFileDevice.cpp

```diff
@@ -385,7 +385,10 @@ bool PcapNgFileReaderDevice::getNextPacket(RawPacket& rawPacket, std::string& pa
 
 	uint8_t* myPacketData = new uint8_t[pktHeader.captured_length];
 	memcpy(myPacketData, pktData, pktHeader.captured_length);
-	if (!rawPacket.setRawData(myPacketData, pktHeader.captured_length, pktHeader.timestamp, static_cast<LinkLayerType>(pktHeader.data_link), pktHeader.original_length))
+	const LinkLayerType linkType = static_cast<LinkLayerType>(pktHeader.data_link);
+	if (linkType == LinkLayerType::LINKTYPE_INVALID)
+		PCPP_LOG_ERROR("Link layer type of packet could not be determined because interface ID exceeds the number of parsed IDBs");
```

Comment:
nit: it's usually a better practice to use `{}` even for one statement. Also, I'd change the error message to something that will be clearer to most users:
```cpp
if (linkType == LinkLayerType::LINKTYPE_INVALID)
{
	PCPP_LOG_ERROR("The link layer type of the packet could not be determined because its interface ID exceeds the number of Interface Description Blocks parsed from the pcapng file");
}
```